### PR TITLE
Add Weather subsystem

### PR DESCRIPTION
**Pull request by agentflare**
Uses a particle generator and a filter layer on top of the map layer to emulate rain and snow.

### CODE REVIEW

File: Live/Classes/WeatherManager.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    m_day = 0;
+}
+
+void WeatherManager::setWeather(WeatherManager::Weather weather) {
+    m_curr_weather = weather;
+    m_weather_layer->removeChild(m_particle_generator);
+    m_weather_layer->removeChild(m_overlay);
+
+    if(weather == RAINING) {
+        m_particle_generator = ParticleRain::createWithTotalParticles(100000);
```
Comment (Jacqueuela) [inline Live/Classes/WeatherManager.cpp]: Use preprocessor macros.

File: Live/Classes/WeatherManager.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        m_overlay = cocos2d::LayerColor::create(Color4B(153, 153, 153, 100));
+        m_overlay->setContentSize(m_weather_layer->getContentSize());
+        m_weather_layer->addChild(m_overlay);
+    }
+};
+
+void WeatherManager::update() {
+    int day = GameTimer::getInstance()->getDaysElapsed();
+    if(day % 10 == 0 && m_day < day) {
```
Comment (Jacqueuela) [inline Live/Classes/WeatherManager.cpp]: Faster day change.

Comment (Jacqueuela) [reply Live/Classes/WeatherManager.cpp]: On reroll don't restart animation.

File: Live/Classes/WeatherManager.cpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+WeatherManager::WeatherManager(Layer* weatherLayer) {
+    m_weather_layer = weatherLayer;
+    m_curr_weather = SUNNY;
+    m_day = 0;
+}
+
+void WeatherManager::setWeather(WeatherManager::Weather weather) {
+    m_curr_weather = weather;
+    m_weather_layer->removeChild(m_particle_generator);
+    m_weather_layer->removeChild(m_overlay);
```
Comment (Jacqueuela) [inline Live/Classes/WeatherManager.cpp]: removeFromParent

File: Live/Classes/MapManager.cpp [hunk 4]
```diff
@@ -3,13 +3,25 @@
 USING_NS_CC;
 
 // Default constructor
-MapManager::MapManager() {
+MapManager::MapManager(Layer* game_layer) {
```
Comment (Jacqueuela) [inline Live/Classes/MapManager.cpp]: I still don't really think you need this, but whatever.

**Review by Jacqueuela — APPROVED** [review]
LGTM

File: Live/Classes/MapManager.cpp [hunk 5]
```diff
@@ -4,17 +4,32 @@ USING_NS_CC;
...
     m_resources = new ResourceLayer();
     m_tile_map->addChild(m_resources, RESOURCE_LAYER_Z_ORDER);
+
+    m_base_layer = Layer::create();;
+    // make base layer the same size as the map
+    m_base_layer->setContentSize(Size(m_map_width, m_map_height));
+    m_base_layer->addChild(m_tile_map, -1);
+
+    auto weather_layer = Layer::create();
+    weather_layer->setContentSize(Size(m_map_width, m_map_height));
```
Comment (sadmansk) [inline Live/Classes/MapManager.cpp]: Move weather layer logic to weather manager contructor.

**Review by sadmansk — CHANGES_REQUESTED** [review]
Lemme know when it's ready.

**Review by sadmansk — APPROVED** [review]
lgtm

**Pull request merged**